src/librustc_mir/transform/check_consts/validation.rs

```diff
@@ -248,12 +248,15 @@ impl Validator<'mir, 'tcx> {
             return;
         }
 
-        // If an operation is supported in miri (and is not already controlled by a feature gate) it
-        // can be turned on with `-Zunleash-the-miri-inside-of-you`.
-        let is_unleashable = O::IS_SUPPORTED_IN_MIRI && O::feature_gate().is_none();
+        // If an operation is supported in miri it can be turned on with
+        // `-Zunleash-the-miri-inside-of-you`.
+        let is_unleashable = O::IS_SUPPORTED_IN_MIRI;
 
         if is_unleashable && self.tcx.sess.opts.debugging_opts.unleash_the_miri_inside_of_you {
-            self.tcx.sess.span_warn(span, "skipping const checks");
+            self.tcx.sess.span_warn(self.tcx.def_span(self.def_id), "skipping const checks");
```

Comment:
Why did you move to reporting on the current item instead of at the span where the const check is being skipped?

Reply:
To make deduplication work. Otherwise we still get one warning per unique span, which can be half a dozen in a 2-line const...

Reply:
ok, so same discussion as above, let's continue there.